client/src/components/App.jsx

```diff
@@ -1,58 +1,93 @@
 import React from 'react';
+import axios from 'axios';
 import Reviews from './Reviews/Reviews.jsx';
 import ProductOverview from './ProductDetails/ProductOverview.jsx';
 import QuestionList from './QandA/QuestionList.jsx';
 import RelatedItems from './RelatedItems/RelatedItems.jsx';
 import NavBar from './Navbar.jsx';
 
-import Reviewsdata from './Reviews/reviewsData.js';
-import QuestionListdata from './QandA/QandAListData.js';
-import relatedItemsdata from './RelatedItems/relatedItemsData.js';
-import productListDummyData from './ProductDetails/sampleData/productListDummyData.js';
-
 class App extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      currentProductID: '39333',
+      currentProductID: '',
       currentStyleID: 234004,
-      productStylesArray: productListDummyData.productStyles.results,
-      currentProduct: productListDummyData.productToDisplay,
+      productStylesArray: [],
+      currentProduct: [],
       reviews: [],
-      questionList: QuestionListdata,
-      relatedItems: relatedItemsdata,
-
+      questionList: [],
+      relatedItems: [],
     };
+    this.cachedProducts = {};
     this.changeCurrentStyle = this.changeCurrentStyle.bind(this);
+  }
 
+  componentDidMount() {
+    this.getProductData('39333');
   }
 
+  getProductData(productId) {
+    let cachedProduct = this.cachedProducts[productId];
+    if (cachedProduct) {
+      let {currentProduct, productStylesArray, reviews, questionList} = cachedProduct;
+      this.setState({
+        currentProductID: productId,
+        currentProduct,
+        productStylesArray,
+        reviews,
+        questionList
+      });
+    } else {
+      axios.get(`./products/${productId}`)
+        .then(({data}) => {
```

Comment:
It may be worth labelling 'data' with something more descriptive, but it's not crucial.

Reply:
'data' is one of the properties that the axios response contains. 
Because that's the only thing I'm going to use, I'm destructuring it.